### DISCUSSION TITLE
Tracking energy out of PISM at ice/ocean boundary

### DISCUSSION THREAD

**Discussion by mankoff** [pism/pism#528] [Q&A]
I'd like to couple PISM ocean to a GCM, and conserve mass and energy across the boundary.

+ I believe `tendency_of_ice_amount_due_to_discharge` are 'icebergs'. It is provided in mass units, but how do I track enthalpy here? I need the potential temperature of this ice mass per grid cell. Is this available? I see `shelfbtemp`, but big tabular icebergs have different temperature than just the bottom layer. Should I use `temppa`?. There are no shelves in Greenland but there are still icebergs, so `shelfbtemp` seems incorrect to use there.

+ Is there an associated ocean temperature variable that goes with `tendency_of_ice_amount_due_to_basal_mass_flux`, or is the freshwater provided by PISM at the phase transition temperature (PTT), not warmed to the ambient temperature? If it's at the PTT, then it seems like I should take the GCM ocean cell that provided the input potential temperature and cool it by some amount such that the PISM water mass at the PTT is warmed to the new (cooler) bulk ambient temperature.

**Comment by citibeth** [comment]
Ken,

On Mon, Dec 18, 2023 at 8:45 PM Ken Mankoff ***@***.***>
wrote:

> I'd like to couple PISM ocean to a GCM, and conserve mass and energy
> across the boundary.
>
>    -
>
>    I believe tendency_of_ice_amount_due_to_discharge are 'icebergs'. It
>    is provided in mass units, but how do I track enthalpy here?
>
> See the code that tracks enthalpy in the ice / atmosphere coupling.  I
believe I "hacked" the PISM code, i.e. created my own PISM classes that
summed up what I wanted when ice was added / removed from the ice sheet.
Each (3d) gridcell has an enthalpy, so it's a matter of tracking how much
ice is added / removed from each gridcell of ice and summing it up.  (I
believe PISM allows for partial gridcells).


>
>    -
>
>    I need the potential temperature of this ice mass per grid cell. Is
>    this available?
>
> I don't understand potential temperature here, that is generally a term
used with atmosphere?  But here we are talking about ice and water.

The state variable used in PISM is enthalpy, not temperature.  Enthalpy is
a conserved quantity whereas temperature is not.  (To see why, imagine an
ice cube at -10C is placed in a large glass of water at 0C and allowed to
come to equilibrium, assuming no energy gain/loss to the surrounding
environment.  Once the system comes to equilibrium, there will be a little
more ice and a little less water, and the ENTIRE SYSTEM will have
temperature 0C.  Thus, temperature is not conserved, and that is why PISM
uses enthalpy, as that is especially important for polythermal glaciers).

https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwi1kd3x6ZqDAxWvHjQIHcDaBHgQFnoECA8QAQ&url=https%3A%2F%2Fglaciers.gi.alaska.edu%2Fsites%2Fdefault%2Ffiles%2Fefgis.pdf&usg=AOvVaw1Bvme2ggbneAb-OX-vaww5&opi=89978449


>    -
>
>    Is there an associated ocean temperature variable that goes with
>    tendency_of_ice_amount_due_to_basal_mass_flux, or is the freshwater
>    provided by PISM at the phase transition temperature (PTT), not warmed to
>    the ambient temperature? If it's at the PTT, then it seems like I should
>    take the GCM ocean cell that provided the input potential temperature and
>    cool it by some amount such that the PISM water mass at the PTT is warmed
>    to the new (cooler) bulk ambient temperature.
>
> In Physics, meltwater is always at 0C (the PTT).  I agree, when water
melts off the ice sheet, it will cool the ocean.

Have you taken into account that ocean melting of Greenland glaciers
happens at the mouth of fjords?  There are formulas to translate the
temperature in the open ocean (which is what the GCM provides) into
temperatures at the head of the fjord (which is what PISM experiences).
There are also formulas for the amount of ice melt based on that
head-of-fjord temperature.  See here:

Wood, M., E. Rignot, I. Fenty, L. An, A. Bjørk, M. van den Broeke, C. Cai,
E. Kane, D. Menemenlis, R. Millan, M. Morlighem, J. Mouginot, B. Noe ̈l, B.
Scheuchl, I. Velicogna, J. K. Willis, and H. Zhang (2021). Ocean forcing
drives glacier retreat in Greenland. Science Advances 7(1), eaba7282.
-- Elizabeth

**Comment by mankoff** [comment]
>> I need the potential temperature of this ice mass per grid cell. Is
>> this available?
>>
> I don't understand potential temperature here, that is generally a
> term used with atmosphere? But here we are talking about ice and
> water.

Ocean has potential temperature.

I understand the enthalpy vs temperature issue.

> In Physics, meltwater is always at 0C (the PTT). I agree, when water
> melts off the ice sheet, it will cool the ocean.

OK, thanks. I will implement this.

> Have you taken into account that ocean melting of Greenland glaciers
> happens at the mouth of fjords?

Yes, I have thought of this, but not yet implemented it. I have a memory of a nice paper I saw years ago (maybe date 1990s) showing temperature gradient along fjords in Alaska, but haven't been able to find it. I think author last name started with a `C`. Still searching...

Thanks for the Wood 2021 paper, that may help.

  -k.
